update.py

```diff
@@ -343,20 +354,29 @@ def main() -> None:
     print("Checking for current bot version and local changes...")
     get_bot_version(git_bin)
 
-    # Check that the current working directory is clean
+    # Check that the current working directory is clean.
+    # -suno is --short with --untracked-files=no
     status_unclean = subprocess.check_output(
-        [git_bin, "status", "--porcelain"], universal_newlines=True
+        [git_bin, "status", "-suno", "--porcelain"], universal_newlines=True
     )
-    if status_unclean:
+    if status_unclean.strip():
+        # TODO: Maybe offering a stash option here would not be so bad...
+        print(
+            "Detected the following files have been modified:\n"
```

Comment:
Maybe we should encode this like the bad install error message? I'll leave that choice up to you.

Reply:
Not sure we use the encoded message any more either. lol

Reply:
Oops I think Jayden wanted that to stay there to deter people from downloading a .zip by just removing it

Reply:
I don't know if I was the one who removed it.  I do remember removing the encoded keys for now long defunct email auth in the config.   
With how the bot pulls its version and checks for the git repo though, I figure folks would have a hard time getting around not being in a git repo these days.

Reply:
Though if it needs to be more clear, not to use the repo archive downloads then we can certainly do that too.

Reply:
It's probably fine I hope lol